outlook/src/taskpane/components/App.tsx

```diff
@@ -106,39 +77,117 @@ export default class App extends React.Component<AppProps, AppState> {
             },
             addRequestCanceller: (canceller: () => void) => {
                 this.requestCancellers.push(canceller);
-            }
+            },
+
+            showTopBarMessage: (enrichmentInfo) => {
+                if (enrichmentInfo)
+                    this.setState({
+                        EnrichmentInfo: enrichmentInfo,
+                        showEnrichmentInfoMessage: true
+                    });
+                else
+                    this.setState({
+                        EnrichmentInfo: new EnrichmentInfo(EnrichmentInfoType.ConnectionError),
+                        showEnrichmentInfoMessage: true
+                    });
+            },
+
+            showHttpErrorMessage: (error?) => {
+                if (error && error.message == '0')
+                {
+                    this.setState({
+                        EnrichmentInfo: new EnrichmentInfo(EnrichmentInfoType.ConnectionError),
+                        showEnrichmentInfoMessage: true
+                    });
+                }
+                else
+                {
+                    this.setState({
+                        EnrichmentInfo: new EnrichmentInfo(EnrichmentInfoType.Other),
+                        showEnrichmentInfoMessage: true
+                    });
+                }
+            },
+
         };
     }
 
-    componentDidMount() {
+    private goToLogin = () => {
         this.setState({
-        isLoading: false,
-        });
+        pageDisplayed: Page.Login
+        })
     }
 
-    goToLogin = () => {
+    private goToMain = () => {
         this.setState({
-        pageDisplayed: Page.Login
+        pageDisplayed: Page.Main
         })
     }
 
-    goToMain = () => {
+    private itemChanged = () => {
+
+    }
+
+    private hideEnrichmentInfoMessage = () => {
         this.setState({
-        pageDisplayed: Page.Main
+            showEnrichmentInfoMessage: false
         })
     }
 
-    itemChanged = () => {
-        this.setState({'doReload': true});
-        //this.forceUpdate();;
+    private hidePartnerCreatedMessage = () => {
+        this.setState({
+            showPartnerCreatedMessage: false
+        })
+    }
+
+    private getMessageBars = () => {
```

Comment:
Side linting note: not sure what are expected guidelines, but sometimes everything is oredered alphabetically, sometimes not ... could be interesting to stick on an alphabetical ordering when in doubt. Just proposing.

Reply:
Class members are ordered according to their access modifiers "public, private..." but I think that it might be interesting to have alphabetical ordering for the members having the same modifiers.